colmena/value_server/proxy.py

```diff
@@ -0,0 +1,199 @@
+                the true object being wrapped
+        """
+        super(ObjectProxy, self).__init__(factory)
+
+    def __reduce__(self):
+        """Helper method for pickling
+        Override `Proxy.__reduce__` so that we only pickle the Factory
+        and not the object itself to reduce size of the pickle.
+        """
+        return ObjectProxy, (self.__factory__,)
+
+    def __reduce_ex__(self, protocol):
+        """See `__reduce__` since we ignore protocol"""
+        return self.__reduce__()
+
+    def async_resolve(self) -> None:
+        if not hasattr(self, '__target__'):
+            self.__factory__.async_resolve()
+
+    def reset_proxy(self) -> None:
+        """Reset wrapped object so that the factory is called on next access"""
+        if hasattr(self, '__target__'):
+            object.__delattr__(self, '__target__')
+
+
+def to_proxy(obj: Any,
+             key: Optional[str] = None,
+             serialization_method: Union[str, SerializationMethod] = SerializationMethod.PICKLE,
+             strict: bool = False
+) -> None:
+    """Put object in value server and return proxy object
+
+    Args:
+        obj (object)
+        key (str, optional): key to use for value server
+        serialization_method (SerializationMethod): serialization method
+        strict (bool): force strict guarentees that Value Server always returns
+            most recent object associated with this key
+
+    Returns:
+        ObjectProxy
+    """
+    if key is None:
+        key = str(id(obj))
+    value_server.server.set(key, obj, serialization_method)
+    return ObjectProxy(Factory(key, serialization_method, strict))
+
+
+def to_proxy_threshold(objs: Union[object, list, tuple, dict],
+                       threshold: Optional[int] = None,
+                       serialization_method: Union[str, SerializationMethod] = SerializationMethod.PICKLE,
+                       strict: bool = False
+) -> Union[object, list, tuple, dict]:
+    """Wrap objects in proxy based on size threshold
+
+    Checks size of object or objects in iterable and wraps all objects with
+    size greater than `threshold` in an `ObjectProxy`
+
+    Args:
+        objs: object, iterable of objects, or dictionary to scan
+        threshold (int): threshold in bytes to determine if an object should
+            be replaced by a proxy
+        serialization_method (str): serialization method to use when placing
+            wrapped objects in the value server
+        strict (bool): force strict guarentees that Value Server always returns
+            most recent object associated with this key
+
+    Returns:
+        Object or iterable with same format as `objs` with any objects of
+        `sys.getsizeof` greater than `threshold` replace by proxies.
+    """
+    def _to_proxy(obj: Any) -> Any:
+        if isinstance(obj, ObjectProxy) or threshold is None:
+            return obj
+        if sys.getsizeof(obj) > threshold:
```

Comment:
Do you know if sizeof works well with tensorflow models or other weird objects? If we run into problems, can we run this "to_proxy" after serializing an object?